lib/commons/models/hive_objects/student.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:hive/hive.dart';
+import 'package:insight/commons/models/hive_objects/settings_object.dart';
+import 'package:insight/commons/models/hive_objects/insight_user_object.dart';
+
+part 'student.g.dart';
+
+@HiveType(typeId: 1)
+class Student {
+  @HiveField(0)
+  late String? registrationNumber;
+
+  @HiveField(1)
+  late String? name;
+
+  @HiveField(2)
+  late String? photoURL;
+
+  @HiveField(3)
+  late bool? isSubscribedToSRM;
+
+  @HiveField(4)
+  late Settings? settings;
+
+  @HiveField(5)
+  late InsightUser? insightUser;
+
+  @override
+  String toString() {
+    return 'Student{registrationNumber: $registrationNumber, \nname: $name, \nphotoURL: $photoURL,\nisSubscribedToSRM: $isSubscribedToSRM, \nsettings: $settings, \ninsightUser: $insightUser}';
+  }
```

Comment:
On second thought it is unnecessary to display all attrs of the object when we try to print it. Registration number is the only uid of a student so that suffices. No need to display other stuff.
```dart
@override
String toString() {
  return "Student{ registrationNumber: $registartionNumber }";
}
```